### PR TITLE
[rllib] Split docs into user and development guide

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Splits out the internal dev docs for RLlib, and add a high level overview on how to define new algorithms and models.

@richardliaw @pcmoritz any suggestions? I didn't add too much of the specifics on sampling / other utils classes since they still seem to be in flux.

### CODE REVIEW

File: python/ray/tune/README.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
-Ray.tune: Efficient distributed hyperparameter search
-=====================================================
+Ray.tune: Hyperparameter Optimization Framework
```
Comment (ericl) [inline python/ray/tune/README.rst]: I felt like there was an overload of adjectives on the docs page, so cleaned this up a bit.

**Comment by richardliaw** [comment]
This looks decent from a first glance. Perhaps it makes sense to ask around for feedback from people who aren't too familiar with the codebase 🙂

**Comment by robertnishihara** [comment]
Also looks reasonable to me.

**Comment by pcmoritz** [comment]
This looks good! The thing that we could provide on top of this that might make it much easier for people to get started implementing algorithms with this is a tutorial style very simple variant of one of the algorithms, like vanilla policy gradients.

**Comment by robertnishihara** [comment]
@pcmoritz that would also be useful to have for tutorials or bootcamps.

**Comment by ericl** [comment]
@pcmoritz I agree! I filed https://github.com/ray-project/ray/issues/1378, but we could also do it for whatever comes up next.

File: python/ray/tune/README.rst (hunk 1 above)
Comment (ericl) [reply python/ray/tune/README.rst]: Suggestions on description welcome...

Comment (richardliaw) [reply python/ray/tune/README.rst]: I think it's good; it definitely declutters the sidebar, and we can work on the teasers and stuff once we have closer integration with the autoscaler and such.

File: doc/source/rllib-dev.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
+RLlib Developer Guide
+=====================
+
+.. note::
+
+    If you want to apply existing RLlib algorithms, first check out the `user docs <http://ray.readthedocs.io/en/latest/rllib.html>`__.
```
Comment (royf) [inline doc/source/rllib-dev.rst]: This guide will take you through steps for implementing a new algorithm in RLlib. To apply existing algorithms already implemented in RLlib, please see the user docs.

File: doc/source/rllib-dev.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+Recipe for an RLlib algorithm
+-----------------------------
+
+Here are the steps for implementing a new algorithm in RLlib:
+
+1. Define an algorithm-specific `Evaluator class <#evaluators-and-optimizers>`__ (the core of the algorithm). Evaluators encapsulate framework-specific components such as the policy and loss functions. For an example, see the `A3C Evaluator implementation <https://github.com/ray-project/ray/blob/master/python/ray/rllib/a3c/a3c_evaluator.py>`__.
+
+
+2. Pick an appropriate `RLlib optimizer class <#evaluators-and-optimizers>`__. Optimizers manage the parallel execution of the algorithm. RLlib provides several built-in optimizers for gradient-based algorithms.
```
Comment (royf) [inline doc/source/rllib-dev.rst]: Advanced algorithms may find it beneficial to implement their own optimizers.

File: doc/source/rllib-dev.rst (hunk 2 above)
Comment (ericl) [reply doc/source/rllib-dev.rst]: Done

File: doc/source/rllib-dev.rst (hunk 3 above)
Comment (ericl) [reply doc/source/rllib-dev.rst]: Done

File: doc/source/rllib.rst [hunk 4]
```diff
@@ -8,28 +8,27 @@ Ray RLlib is a reinforcement learning library that aims to provide both performa
...
 
 - Composability
-    - Integration with the `Ray.tune <http://ray.readthedocs.io/en/latest/tune.html>`__ hyperparam tuning tool
+    - Integration with the `Ray.tune <tune.html>`__ hyperparam tuning tool
     - Support for multiple frameworks (TensorFlow, PyTorch)
     - Scalable primitives for developing new algorithms
     - Shared models between algorithms
 
-You can find the code for RLlib `here on GitHub <https://github.com/ray-project/ray/tree/master/python/ray/rllib>`__, and the NIPS symposium paper `here <https://drive.google.com/open?id=1lDMOFLMUQXn8qGtuahOBUwjmFb2iASxu>`__.
+You can find the code for RLlib `here on GitHub <https://github.com/ray-project/ray/tree/master/python/ray/rllib>`__, and the NIPS symposium paper `here <https://arxiv.org/abs/1712.09381>`__.
```
Comment (richardliaw) [inline doc/source/rllib.rst]: `NIPS 2017 Deep Reinforcement Learning Symposium` would be a better reference

**Review by richardliaw — APPROVED** [review]

**Pull request merged**